### PR TITLE
Add calculator event handler

### PR DESCRIPTION
**Pull request by bimbolabuari**
## Math Magician Calculator 🙂 

 > This PR adds a Microverse project on working with React events for a calculator app, that demonstrates the knowledge of React components, event handling, life cycle methods, lifting state up, states, props.

## Live Preview 🎉  

[Live Demo Link](https://deploy-preview-4--bimbola-mv-math-magician.netlify.app/)

### CODE REVIEW

File: src/logic/Calculate.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      }
+      return { total: `${obj.next}.` };
+    }
+    return { total: '0.' };
+  }
+
+  if (buttonName === '=') {
+    if (obj.next && obj.operation) {
+      return {
+        total: operate(obj.total, obj.next, obj.operation),
```
Comment (Davidosky007) [inline src/logic/Calculate.js:71]: * I get errors when i perform divide by zero operation (ie 7 / 0) 
image

File: src/logic/Calculate.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import operate from './operate';
+
+function isNumber(item) {
+  return !!item.match(/[0-9]+/);
+}
+
+
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
```
Comment (Davidosky007) [inline src/logic/Calculate.js:17]: * Kindly fix the `AC` button as it doesn't clear the input field unless i refresh the browser 🙏

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @bimbolabuari, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/displayResult.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          }
+          return (
+           <div className="calculator-screen">{display}</div>
+           
+          );
+      }
+    }
+    
+    DisplayResult.propTypes = {
+      total: PropTypes.string,
```
Comment (Davidosky007) [inline src/components/displayResult.js]: * Seems while working on the requested changes another error was created kindly correct the prop types for total it expects a string but gets a number instead 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @bimbolabuari, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @bimbolabuari, 

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**